### DISCUSSION TITLE
Unexpected behavior of trapped ion's motional mode (phonon) heating simulation

### DISCUSSION THREAD

**Discussion by kymikim0401** [qutip/qutip#2515] [Q&A]
Hello, I'm a novice at qutip simulation.

I was recently trying to use qutip to simulate the heating effect of a trapped ion by defining a jump operator defined with heating rate. 

Below is the code of my simulation:

```python
from qutip import * 
from qutip_utils import *

exp_config = {
    'lamb_dicke' : 0.095043,
    'n_init' : 0.1, # initial number of phonons of a trapped ion 
    'carrier_Rabi' : 2 * np.pi * 120 * 1000,
    'dim' : 500, # allowed phonon distribution dimension (i.e. we are interested in p(n=0) to p(n=499))
    'trap_freq' : 2 * np.pi * 1.02 * 1e6,
    'gamma' : 1000} # heating rate

max_N = exp_config['dim']
avg_n = exp_config['n_init']
ldf = exp_config['lamb_dicke']
rabi = exp_config['carrier_Rabi']
heating_rate = exp_config['gamma'] 

phonon_dm = thermal_dm(max_N, avg_n)
dstate = basis(2, 0)
ustate = basis(2, 1)
H0 = tensor(qeye(2), qeye(max_N))
rho0 = tensor(ket2dm(dstate), phonon_dm)
result = []
heating_time = 5e-3
times = np.linspace(0, heating_time, 100)
up_proj = tensor(ket2dm(ustate), qeye(max_N))
down_proj = tensor(ket2dm(dstate), qeye(max_N))
num_proj = tensor(qeye(2), create(max_N)*destroy(max_N))

jump_heat = np.sqrt(heating_rate)*tensor(qeye(2), create(max_N))
jump_ops = [jump_heat]

initial_state = rho0
H_heating = [H0]
motion_data = mesolve(H_heating, initial_state, times, jump_ops, num_proj, options={'store_states': True})
final_state_phonon = motion_data.states[-1].ptrace(1)
result.append(motion_data.expect[0][-1])

print(f'answer = {exp_config["gamma"] * heating_time}')
```

I have run the simulation where the heating lasts for heating_time = 5e-3 (5ms), which means the number of phonons increased should be approximately heating_rate times heating time ~= 5. However, If I run the simulation and get the expectation value of phonons after the heating (motion_data.expect[0][-1] part), I get 155. 

Below is the gif that shows the change of phonon probability distribution of above simulation: 

![output](https://github.com/user-attachments/assets/3810a937-01ca-4bf4-8dab-07a5d251af73)

where Y axis is the probability value and X axis is the harmonic state (n) of phonon up to dim=500. At the very last of the simulation, there is an instantaneous spike at n~= 500, which seems wrong. 

Is there anything I did wrong with my code? I assume modifying the defined hamiltonian could be one way of fixing this anomaly, but I'm not sure. Please help :(

**Comment by Ericgig** [comment]
The code itself seems fine. You could try changing the tolerance in options: `atol`, `rtol`, `nsteps`. Also using `"normalize_output"=False` remove a step that could hide numerical error and could help finding the issue.

**Reply by kymikim0401** [reply]
Thanks for the reply.

I have tried options = {'atol': 1e-10, 'rtol': 1e-9, 'nsteps': 100000}, but still I get the results close to 155 :(

**Comment by nwlambert** [comment]
just a quick comment, do you have some reason to expect the occupation to grow to heating rate * final-time? i guess the analytical solution to the problem you set up (an oscillator with only heating, no emission) is just exponential growth with time, which fits what you see in the simulation.   maybe you also need to add some competing cooling/spontaneous emission as well?

**Reply by nwlambert** [reply]
also in your plot of the populations of different states, the weird jump at the end is probably just from the normal error you get from truncating the fock space. it shouldn't be affecting your results yet, with <n>~ 155 and a cut-off of 500, but it will eventually if you evolve for longer times

**Comment by kymikim0401** [comment]
Thank you all for the comments. I think I have solved the issue by adding a conjugate heating term, jump_heat_conj = np.sqrt(heating_rate)*tensor(qeye(2), destroy(max_N)).
